src/main/java/io/vertx/proton/impl/ProtonTransport.java

```diff
@@ -284,4 +284,9 @@ public void handle(Long event) {
       }
     }
   }
+
+  ProtonTransport setIdleTimeout(int idleTimeout) {
+    this.transport.setIdleTimeout(idleTimeout);
```

Comment:
FWIW, part of where I mislead myself here was that while there are cases it wouldn't have worked if the new setHeartbeat option was on the connection object, that isn't a problem when its on the server options since the option effect is then applied much earlier as a result.

This setter can now be deleted as the updates remove the need for it.